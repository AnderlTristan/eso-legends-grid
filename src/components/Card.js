import React from 'react'
import styled from 'styled-components'

export const Card = ({name, text, type, setName, imageUrl}) => {
    return (
        <StyledCard className="card">
            <h2 className="card__title">{name}</h2>
            <div className="card__image"><img src={imageUrl}/></div>
            <div className="card__detail-wrapper">
                <div className="card__accent-left">{type}</div>
                <div className="card__accent-right">{setName}</div>
                <div className="card__description">{text}</div>
            </div>
        </StyledCard>
    )
}

const StyledCard = styled.div`
    max-width: 250px;
    height: auto;
    min-height: 600px;
    padding: 15px;
    margin-top: 40px;
    margin-left: 15px;
    margin-right: 15px;
    border-radius: 15px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    background: #e0e5eb;
    color: #222;
    box-shadow: 17px 17px 27px #acb0b5, -17px -17px 27px #ffffff;
    text-align: center;

    img {
        width: 90%;
        height: auto;
    }
    div {
        font-family: 'lato', sans-serif;
        display: block;
        margin: 5px auto;
    }

    .card__detail-wrapper {
        display: flex;
        flex-direction: row;
        text-align: center;
        flex-wrap: wrap;
        align-items: center;
        width: 100%;
    }
    .card__image {
        min-height: 375px;
    }
    .card__image {
        text-align: center;
    }
    .card__title {
        font-family: 'Cardo', serif;
        display: block;
        margin: 5px auto;
        text-transform: uppercase;
        width: 100%;
        min-height: 55px;
        display: flex;
        justify-content: center;
        align-items: center;
    }
    .card__description {
        width: 100%;
        min-height: 90px;
    }
    .card__accent-right,
    .card__accent-left {
        font-family: 'Cardo', serif;
        width: calc(50% - 24px);
        border-radius: 15px;
        padding: 5px 0;
        background: linear-gradient(145deg, #caced4, #f0f5fb);
        box-shadow:  20px 20px 60px #bec3c8, -20px -20px 60px #ffffff;
        font-weight: bold;
        min-height: 50px;
        display: flex;
        align-items: center;
        justify-content: center;
        text-align: center;
    }
`;