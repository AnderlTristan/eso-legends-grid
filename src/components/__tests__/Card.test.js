import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Card } from '../Card';

afterEach(() => {
    cleanup();
});

test('should render card component', () => {
    const cardProps = {
        "name": "Adoring Fan",
        "set": {
            "name": "Core Set",
        },
        "text": "Prophecy, Guard. Last Gasp: Adoring Fan will return. Adoring Fan is immune to Silence.",
        "imageUrl": "https://images.elderscrollslegends.io/cs/adoring_fan.png",
        "type": "Creature"
    }

    render(
        <Card 
            name={cardProps.name} 
            imageUrl={cardProps.imageUrl}
            type={cardProps.type}
            setName={cardProps.set.name}
            text={cardProps.text} 
        />
    );

    const cardElement = screen.getByTestId('card');
    expect(cardElement).toBeInTheDocument();
    expect(cardElement).toHaveTextContent("Adoring Fan");  
});

test('matches snapshot', () => {
    const cardProps = {
        "name": "Adoring Fan",
        "set": {
            "name": "Core Set",
        },
        "text": "Prophecy, Guard. Last Gasp: Adoring Fan will return. Adoring Fan is immune to Silence.",
        "imageUrl": "https://images.elderscrollslegends.io/cs/adoring_fan.png",
        "type": "Creature"
    };

    const tree = renderer.create(        
        <Card 
            name={cardProps.name} 
            imageUrl={cardProps.imageUrl}
            type={cardProps.type}
            setName={cardProps.set.name}
            text={cardProps.text} 
        />
    );
    
    expect(tree).toMatchSnapshot();
});