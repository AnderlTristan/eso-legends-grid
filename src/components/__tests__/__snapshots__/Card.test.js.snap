// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdnxRM hEMIaW card"
  data-testid="card"
>
  <h2
    className="card__title"
  >
    Adoring Fan
  </h2>
  <div
    className="card__image"
  >
    <img
      src="https://images.elderscrollslegends.io/cs/adoring_fan.png"
    />
  </div>
  <div
    className="card__detail-wrapper"
  >
    <div
      className="card__accent-left"
    >
      Creature
    </div>
    <div
      className="card__accent-right"
    >
      Core Set
    </div>
    <div
      className="card__description"
    >
      Prophecy, Guard. Last Gasp: Adoring Fan will return. Adoring Fan is immune to Silence.
    </div>
  </div>
</div>
`;
